pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -551,33 +553,40 @@ func (c *Cluster) initMonIPs(mons []*monConfig) error {
 	return nil
 }
 
-func (c *Cluster) assignMons(mons []*monConfig) error {
-	// when monitors are scheduling below by invoking scheduleMonitor() a canary
-	// deployment and optional canary PVC are created. in order for the
-	// anti-affinity rules to be effective, we leave the canary pods in place
-	// until all of the canaries have been scheduled. only after the
-	// monitor/node assignment process is complete are the canary deployments
-	// and pvcs removed here.
-	canaryCleanup := []SchedulingResult{}
-	defer func() {
-		for _, result := range canaryCleanup {
-			logger.Infof("assignmon: cleaning up canary deployment %s and canary pvc %s", result.CanaryDeployment, result.CanaryPVC)
-			if result.CanaryDeployment != "" {
-				if err := k8sutil.DeleteDeployment(c.context.Clientset, c.Namespace, result.CanaryDeployment); err != nil {
-					logger.Infof("assignmon: error deleting canary deployment %s. %v", result.CanaryDeployment, err)
-				}
+// Delete mon canary deployments (and associated PVCs) using deployment labels
+// to select this kind of temporary deployments
+func (c *Cluster) removeCanaryDeployments() {
+	canaryDeployments, err := k8sutil.GetDeployments(c.context.Clientset, c.Namespace, "app=rook-ceph-mon, mon_type=canary")
+	if err != nil {
+		logger.Errorf("error trying to get the list of canary deployments: %v", err)
+	} else {
+		for _, canary := range canaryDeployments.Items {
+			logger.Infof("removeCanaryDeployments: cleaning up canary deployment %q and canary pvc %q", canary.Name, canary.Labels["canary_pvc_name"])
```

Comment:
Rook doesn't elsewhere use the convention of prefacing logs with the function name.
```suggestion
			logger.Infof("cleaning up canary deployment %q and canary pvc %q", canary.Name, canary.Labels["canary_pvc_name"])
```

Reply:
True. Removed the function name from the log message.